### DISCUSSION TITLE
Initial thoughts and rants

### DISCUSSION THREAD

**Discussion by JPCarvalhinho** [rhinstaller/anaconda-webui#34] [General]
First of all congrats. It is a very nice step up from the previous installer. And without further ado I will try to categorize it in the title...
Installed it on Boxes in Fed47 on an Imac

1. Welcome screen
1.0 Main title "Anaconda Installer" should be more descriptive "Anaconda Installer for Fedora 37 Desktop" (if you are a multiple OS user, believe me this is never too much information)
1.1- [lang] Unfinished translation to PT (DUH! still under development)
1.2- [lang] [Keyb] Impossible to choose Keyboard. Should not be necessary, as the account creation only happens on first launch, and partition naming should not contain weird chars, not sure if necessary or if this is for the best... 
1.3- [UI] You should never leave important information for "optional content areas" like information boxes. You should not use the information box to pass important system information.
1.4- [UI] Language list - Include more entries visible if available screen real estate below. User hates to navigate through thousands of entries on a 6 line basis... if you can display more than 6 lines... display more than 6 lines. 
1.5- [UI] Buttons... buttons... this is not consensual but... the mental state of the user includes the left and right positioning as moving forward or back, courtesy of years using browsers. 

See screenshot "screen1.png" for alternate texts:
![screen1](https://user-images.githubusercontent.com/4802489/203421133-a3eaf0b4-22c5-4aa8-bd17-d78bfd1453e7.png)

you can also Switch the order of the buttons to 
Back (secondary), Next (primary) and Quit (Grey/red background on hover), as the Quit button is not "competing" for the same level of attention as the previous two. 
See "Butons.png

![buttons](https://user-images.githubusercontent.com/4802489/203421114-e18a9088-4a4e-47d5-a7ae-c28266d509b7.png)

2- Partition screen. 
I am assuming a lot of options are missing here, like partition configuration and all, so will not focus on features, but UI only, and 
2.1  [UI] On the onboarding text, take ownership of the decisions done by Anaconda. The new user will be thankfull and feel guided by someone more expert.
2.2 [UI] "Local Santdard disks" -> "Detected Local Storage Devices" 
2.3 [UI] Button "Detect disks"-> "Re-Detect" or "a button with the refresh icon" (Circular arrow)

Check "screen2.png"
![screen2](https://user-images.githubusercontent.com/4802489/203421156-1e00a2dc-75dd-4b9a-8319-f0bfb6d7c370.png)

2.4 On "Learn more about your storage options" fly in you mention "refresh the page using the refresh button" but there is no refresh button, as browser controls are not present. 
Should say  "refresh the devices list using the Detect Disks"

Check screen3.png
![screen3](https://user-images.githubusercontent.com/4802489/203421174-c37cdde7-debf-4dab-98ae-c991658f8059.png)

I am no expert, but I can tell you that this is one of the most complex steps for the newbie. The user just decided to erase the full disk, and delete all his contents to install Fedora. Just help him by owning your (Anaconda) decisions. 
Also I really want to know how you will solve the partitioning options, as it is usually the most complex ideas behind the instalation.

**Comment by JPCarvalhinho** [comment]
I saw from other discussions that creating users and passwords will be a thing for Anaconda also. IF so, point 1.2 becomes mandatory.

**Comment by jkonecny12** [comment]
Hi, this is great feedback thanks a lot for that.

As for partitioning, it is a big topic and one of the most complex thing Anaconda is doing. Right now, we are working on a way to get feedback to be able to decide how users would like to have the partitioning solved. Stay tuned if you want to give us feedback for this :).

**Comment by JPCarvalhinho** [comment]
For me, even before showing the partition table, there should be some kind of toggle with 3 positions:
1- Put the complete system in one partition (default partition table)
2- Please separate my personal files from the system (divides space between "Home" and all the rest in two separate partitions)
3- let me configure it in detail (go bonkers with personalization)

From my limited experience, #2 is the only "different" option a new user would possibly want, and I never got to do it myself because #3 is always too complex for me.

**Comment by MaxGarf** [comment]
I have to things I'd like to see modified:

1. Provide an option to select a different keyboard layout than the chosen system language might imply. I for example use a notebook with German keyboard layout but want English as system language. This becomes important when the possibility to set passwords for disc encryption and/or user accounts is implemented: If the keyboard layout is wrong and the passwords are obscured by only printing dots/stars you get a high risk of users getting stuck with non working passwords later on because the chosen passwords do not correspond to the physical keys pressed.
2. On the progress screen during system installation I suggest a less dominant indicator that the system is doing something. Why not make the small icon of the actual process step being active do something instead of the giant rotating circle on top. The giant circle attracts the eye away from the actual information how far the installation has progressed.
